eng/pipelines/templates/stages/source-build-and-validate.yml

```diff
@@ -207,45 +207,45 @@ stages:
               downloadFilePatterns: assets/Release/**/*.tar.gz
               copyDestination: $(Build.SourcesDirectory)/artifacts
               ${{ if eq(leg.distro, 'centos') }}:
-                artifactName: ${{ format(leg.buildNameFormat, variables.centOSStreamName) }}_${{ leg.targetArchitecture}}_Artifacts
+                artifactName: ${{ format(leg.buildNameFormat, variables.centOSStreamName) }}_${{ leg.targetArchitecture}}_BuildLogs_Attempt1
```

Comment:
I guess the obvious reaction here is whether the Attemp1 hardcode will cause issues with reruns?

Reply:
Yeah I'd rather have these logs copied into the artifacts directory instead of logs being used as inputs to later jobs.

This change is also due to this PR fixing https://github.com/dotnet/source-build/issues/5072

Reply:
I think we should have the files that are consumed here put in the Artifacts payload and download them from there instead of putting them in the build logs payload.